docs/reference/templates.rst

```diff
@@ -172,7 +172,7 @@ can specify the templates to use in the ``Admin`` service definition:
                     - App\Entity\Post
                     - ~
                 calls:
-                    - [setTemplate, ['edit', '@App/PostAdmin/edit.html.twig']]
+                    - [setTemplate, ['edit', '@@App/PostAdmin/edit.html.twig']]
```

Comment:
I thought the change you was proposing was about the template paths for non registered paths. The path `@App/PostAdmin/edit.html.twig` is fine if the `@App` namespace [is registered](https://symfony.com/doc/current/templating/namespaced_paths.html#registering-your-own-namespaces). You don't need to escape the `@` character, since their meaning is correct in that case.
```suggestion
                    - [setTemplate, ['edit', 'PostAdmin/edit.html.twig']]
```
BTW, it would be fine if a reference note to the registered variant is present here, to avoid confusions.

Reply:
There is maybe two changes to made.

I'm actually using Symfony 3 in my project. And in the symfony 3 doc there is the same documentation https://symfony.com/doc/3.x/bundles/SonataAdminBundle/reference/templates.html

In my case `@@App/PostAdmin/edit.html.twig` is the best solution

Reply:
If in your scenario `@App` is a registered namespace (provided by the bundle related logic), there is nothing to escape. Using `@@App/PostAdmin/edit.html.twig` you are indicating that you have a real directory named `@App/` at your filesystem. If that is the case, there is nothing to change here.

Reply:
`@` seems to be a keyword to reference a service.

That's why I have in my project, cf issue #but 5577
```
The service app.admin.post.template_registry" has a dependency on a non-existent service "App/PostAdmin/edit.html.twig".
```
Using `@@App/PostAdmin/edit.html.twig` reference to the string `@App/PostAdmin/edit.html.twig` which is the way twig files are imported.

Reply:
I find the same issue on stackoverflow
https://stackoverflow.com/questions/53046391/symfony-3sonata-admin-settemplate-problem
https://stackoverflow.com/questions/48750338/sonata-admin-bundle-custom-template

`@` is for service injection ; that's why it has to be escaped.

Reply:
@phansys Hi, did you have time to read my answer and possibly review your point of view?

The actual doc do not for work for both Sf3 and Sf4.
You proposal works for Sf4 but not Sf3. Mine works for both.

Reply:
I'm sorry, I couldn't check this issue yet; but if it really exists, the solution you're proposing seems like a workaround. The real solution should be to interpret the Twig path syntax correctly.

Reply:
> The real solution should be to interpret the Twig path syntax correctly.

Indeed, that would be great.